discovery-infra/test_infra/controllers/node_controllers/terraform_controller.py

```diff
@@ -4,46 +4,53 @@
 import os
 import shutil
 import uuid
+import warnings
 from typing import List
 
 from munch import Munch
 from test_infra import consts, utils, virsh_cleanup
 from test_infra.controllers.node_controllers.libvirt_controller import LibvirtController
 from test_infra.controllers.node_controllers.node import Node
+from test_infra.helper_classes.config import BaseTerraformConfig
 from test_infra.tools import static_network, terraform_utils
 
 
 class TerraformController(LibvirtController):
 
-    def __init__(self, **kwargs):
-        super().__init__(**kwargs)
-        self.cluster_suffix = kwargs.get('cluster_suffix', self._get_random_name())
-        self.cluster_name = kwargs.get('cluster_name', f'{consts.CLUSTER_PREFIX}' + "-" + self.cluster_suffix)
-        self.network_name = kwargs.get('network_name', consts.TEST_NETWORK) + self.cluster_suffix
-        self.network_conf = kwargs.get('net_asset')
-        self.platform = kwargs.get('platform', consts.Platforms.BARE_METAL)
-        self.cluster_domain = kwargs.get('base_domain', consts.DEFAULT_BASE_DNS_DOMAIN)
-        self.ipv6 = kwargs.get('ipv6')
-        self.params = self._terraform_params(**kwargs)
-        tf_folder = kwargs.get('tf_folder')
-        self.tf_folder = tf_folder if tf_folder else self._create_tf_folder()
-        self.image_path = kwargs["iso_download_path"]
-        self.bootstrap_in_place = kwargs.get('bootstrap_in_place', False)
+    def __init__(self, config: BaseTerraformConfig):
+        super().__init__(config.private_ssh_key_path)
+        self.config = config
+        self.cluster_suffix = self._get_cluster_suffix(config.cluster_name)
+        self.cluster_name = config.cluster_name
+        self.network_name = config.network_name + self.cluster_suffix
+        self.tf_folder = config.tf_folder or self._create_tf_folder(self.cluster_name, config.platform)
+        self.params = self._terraform_params(**config.get_all())
         self.tf = terraform_utils.TerraformUtils(working_dir=self.tf_folder)
         self.master_ips = None
 
-    def _create_tf_folder(self):
-        tf_folder = utils.get_tf_folder(self.cluster_name)
+    def _get_cluster_suffix(self, cluster_name: str):
+        try:
+            cluster_suffix = cluster_name.split("-")[-1]
+            assert len(cluster_suffix) == 8
+        except (IndexError, AssertionError):
+            cluster_suffix = self._get_random_name()
```

Comment:
What is this new logic? Please add a docstring

Reply:
The suffix was generated twice, on creating cluster name and on creating the nodes.
Now the network, Cluster and nodes will have the same suffix